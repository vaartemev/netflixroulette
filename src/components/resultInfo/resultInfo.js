import React from 'react';
import PropTypes from 'prop-types';
import './resultInfo.scss';

export const ResultInfo = ({ foundedMovies }) => {
  return (
    <nav className="additional">
      <span className="founded">{foundedMovies} movies founded</span>
      <div className="sort">
        <span className="sort__title">Sort by</span>
        <a href="" className="sort__toggle sort__toggle-active sort__release">
          release date
        </a>
        <a href="" className="sort__toggle sort__rating">
          rating
        </a>
      </div>
    </nav>
  );
};

ResultInfo.propTypes = {
  foundedMovies: PropTypes.number.isRequired
};
