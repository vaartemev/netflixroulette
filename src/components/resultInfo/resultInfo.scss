.additional {
	height: 50px;
	line-height: 50px;
	padding: 0 70px;
	background-color: #f3f3f3;
	font-weight: bold;
	color: #656565;
	display: flex;
	justify-content: space-between
}

.sort {
	&__title {
		margin-right: 20px;
		text-transform: uppercase;
		font-size: 14px;
	}
}