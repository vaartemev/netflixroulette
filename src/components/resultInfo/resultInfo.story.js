import React from 'react';
import { storiesOf } from '@storybook/react';
import { ResultInfo } from './resultInfo';

storiesOf('ResultInfo', module).add('ResultInfo', () => (
  <ResultInfo foundedMovies={8} />
));
