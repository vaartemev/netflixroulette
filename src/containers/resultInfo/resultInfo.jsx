import React from 'react';
import { withRouter } from 'next/router';
import { useSelector } from 'react-redux';
import { setSortFilter } from '../../actions';
import { ButtonType } from '../../components';
import { RadioButton } from '../radioButton';
import './resultInfo.scss';

const sortBy = [
  { key: 'release_date', title: 'release date' },
  { key: 'rating', title: 'rating' },
];

export const ResultInfo = withRouter(() => {
  const { foundedMovies, genre } = useSelector(state => ({
    foundedMovies: state.movie.get('movies'),
    genre: state.movie.get('searchQuery'),
  }));

  return (
    <nav className="additional">
      {true ? (
        <span className="founded">Movies by {genre} genre</span>
      ) : (
        <>
          <span className="founded">{foundedMovies.length} movies founded</span>
          <div className="sort">
            <span className="sort__title">Sort by</span>
            <RadioButton
              type={ButtonType.Sort}
              filters={sortBy}
              filter={setSortFilter}
            />
          </div>
        </>
      )}
    </nav>
  );
});
