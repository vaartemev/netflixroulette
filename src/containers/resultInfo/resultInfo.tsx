import React from 'react';
import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import { setSortFilter } from '../../actions';
import { ButtonType } from '../../components';
import { RadioButton } from '../radioButton';
import { resultSelector } from '../../selectors';
import './resultInfo.scss';

interface SortTypes {
  key: string;
  title: string;
}
const sortBy: SortTypes[] = [
  { key: 'release_date', title: 'release date' },
  { key: 'rating', title: 'rating' },
];

export const ResultInfo = () => {
  const { movies, genre } = useSelector(resultSelector);

  const router = useRouter();
  const { id } = router.query;

  return (
    <nav className="additional">
          {id ? (
        <span className="founded">Movies by {genre} genre</span>
      ) : (
              <>
          <span className="founded">{movies.length} movies founded</span>
          <div className="sort">
            <span className="sort__title">Sort by</span>
            <RadioButton
                type={ButtonType.sort}
                filters={sortBy}
              filter={setSortFilter}
            />
          </div>
        </>
      )}
    </nav>
  );
};
