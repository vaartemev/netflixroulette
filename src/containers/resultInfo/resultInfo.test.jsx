import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { ResultInfo } from './resultInfo';
import { configureStore } from '../../store';

const store = configureStore();

describe('ResultInfo', () => {
  const output = mount(
    <Router>
      <Provider store={store}>
        <ResultInfo />
      </Provider>
    </Router>,
  );
  it('Render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
