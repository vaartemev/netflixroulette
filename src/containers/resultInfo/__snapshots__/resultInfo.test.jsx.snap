// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultInfo Render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <withRouter()>
        <Component
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <nav
            className="additional"
          >
            <span
              className="founded"
            >
              0
               movies founded
            </span>
            <div
              className="sort"
            >
              <span
                className="sort__title"
              >
                Sort by
              </span>
              <RadioButton
                filter={[Function]}
                filters={
                  Array [
                    Object {
                      "key": "release_date",
                      "title": "release date",
                    },
                    Object {
                      "key": "rating",
                      "title": "rating",
                    },
                  ]
                }
              >
                <Button
                  checked={true}
                  handleOnClick={[Function]}
                  key="release_date"
                  text="release date"
                >
                  <button
                    className="button button__sort-active button__sort"
                    onClick={[Function]}
                    type="button"
                  >
                    release date
                  </button>
                </Button>
                <Button
                  checked={false}
                  handleOnClick={[Function]}
                  key="rating"
                  text="rating"
                >
                  <button
                    className="button button__sort"
                    onClick={[Function]}
                    type="button"
                  >
                    rating
                  </button>
                </Button>
              </RadioButton>
            </div>
          </nav>
        </Component>
      </withRouter()>
    </Provider>
  </Router>
</BrowserRouter>
`;
